src/bsoncxx/lib/bsoncxx/private/immortal.hh

```diff
@@ -0,0 +1,49 @@
+// Copyright 2009-present MongoDB, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <bsoncxx/v1/detail/macros.hpp>
+
+#include <new>
+
+namespace bsoncxx {
+
+template <typename T>
+class immortal {
```

Comment:
Add a doc comment on what this class template does any why it might want to be used, and how it should not be used (e.g. declaring a non-`static` `immortal<T>` is potentially bad).